oauth2_provider/tests/test_authorization_code.py

```diff
@@ -908,6 +909,139 @@ def test_code_exchange_succeed_when_redirect_uri_match(self):
         self.assertEqual(content['scope'], "read write")
         self.assertEqual(content['expires_in'], oauth2_settings.ACCESS_TOKEN_EXPIRE_SECONDS)
 
+    def test_pkce(self):
+        self.client.login(username="test_user", password="123456")
+        hashed_data = base64.urlsafe_b64encode(hashlib.sha256(b'secret').hexdigest())
+
+        # retrieve a valid authorization code
+        authcode_data = {
+            'client_id': self.application.client_id,
+            'state': 'random_state_string',
+            'scope': 'read write',
+            'redirect_uri': 'http://example.it?foo=bar',
+            'response_type': 'code',
+            'allow': True,
+            'code_challenge': hashed_data
+        }
+        response = self.client.post(reverse('oauth2_provider:authorize'), data=authcode_data)
+        query_dict = parse_qs(urlparse(response['Location']).query)
+        authorization_code = query_dict['code'].pop()
+
+        # exchange authorization code for a valid access token
+        token_request_data = {
+            'grant_type': 'authorization_code',
+            'code': authorization_code,
+            'redirect_uri': 'http://example.it?foo=bar',
+            'code_challenge': 'secret'
```

Comment:
The key for the token request should be `code_verifier`: https://tools.ietf.org/html/rfc7636#section-4.5